# HW04 – Recommender Systems Which movie should we watch next?

#### It’s cold and rainy and you are looking to have a day of movie binging with your family.
#### You have decided that the first movie that you will watch is Toy Story but would like a
#### recommendation on the next movie. Since you are in a good mood and would like to
#### continue to watch fun movies, you would like to filter movies based on comedies.
#### Which comedy movie should you watch next?
#### Let’s create a recommender system to find out.

### By Donovan Manogue

### 3. Import Libraries

In [7]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
# In this section I went ahead and imported the basic libraries needed

### 4. Import Data from a URL

In [16]:
df_movies = pd.read_csv('Homework4_Data/movies.csv', index_col = None , header= 0)
df_genres = pd.read_csv('Homework4_Data/movies_genres.csv', index_col = None , header= 0)
df_ratings= pd.read_csv('Homework4_Data/ratings.csv', index_col = None , header= 0)

In [18]:
df_genres.head()

id     Genre1    Genre2          Genre3 Genre4 Genre5 Genre6 Genre7 Genre8
0   2      Drama     Crime             NaN    NaN    NaN    NaN    NaN    NaN
1   3      Drama    Comedy             NaN    NaN    NaN    NaN    NaN    NaN
2   5      Crime    Comedy             NaN    NaN    NaN    NaN    NaN    NaN
3   6     Action  Thriller           Crime    NaN    NaN    NaN    NaN    NaN
4  11  Adventure    Action  ScienceFiction    NaN    NaN    NaN    NaN    NaN

In [20]:
# Importing the Data from Data Files

### 5. Data preparation to filter based on comedies

#### **OBJECTIVE**: As is always the case, prepping your data is hard. 
#### For this homework, before we can start conducting recommender systems, we need to filter the movies based on it being a comedy.

### Section 5a: Create df_genres_pivot

In [47]:
df_genres_melt = pd.melt(df_genres, id_vars =['id'],
                        value_vars=['Genre1','Genre2','Genre3','Genre4','Genre5','Genre6','Genre7','Genre8'])

In [49]:
df_genres_melt.head()

id variable      value
0   2   Genre1      Drama
1   3   Genre1      Drama
2   5   Genre1      Crime
3   6   Genre1     Action
4  11   Genre1  Adventure

In [51]:
df_genres_melt['Count'] = 1

In [55]:
df_genres_melt.head()

id variable      value  Count
0   2   Genre1      Drama      1
1   3   Genre1      Drama      1
2   5   Genre1      Crime      1
3   6   Genre1     Action      1
4  11   Genre1  Adventure      1

In [59]:
df_genres_pivot = pd.pivot_table(df_genres_melt, index='id', columns ='value', values='Count')

In [61]:
df_genres_pivot =df_genres_pivot.reset_index()
df_genres_pivot.head()

value  id  Action  Adventure  Animation  Comedy  Crime  Documentary  Drama  \
0       2     NaN        NaN        NaN     NaN    1.0          NaN    1.0   
1       3     NaN        NaN        NaN     1.0    NaN          NaN    1.0   
2       5     NaN        NaN        NaN     1.0    1.0          NaN    NaN   
3       6     1.0        NaN        NaN     NaN    1.0          NaN    NaN   
4      11     1.0        1.0        NaN     NaN    NaN          NaN    NaN   

value  Family  Fantasy  ...  History  Horror  Music  Mystery  Romance  \
0         NaN      NaN  ...      NaN     NaN    NaN      NaN      NaN   
1         NaN      NaN  ...      NaN     NaN    NaN      NaN      NaN   
2         NaN      NaN  ...      NaN     NaN    NaN      NaN      NaN   
3         NaN      NaN  ...      NaN     NaN    NaN      NaN      NaN   
4         NaN      NaN  ...      NaN     NaN    NaN      NaN      NaN   

value  ScienceFiction  TVMovie  Thriller  War  Western  
0                 NaN      NaN       NaN  NaN      NaN  
1                 NaN      NaN       NaN  NaN      NaN  
2                 NaN      NaN       NaN  NaN      NaN  
3                 NaN      NaN       1.0  NaN      NaN  
4                 1.0      NaN       NaN  NaN      NaN  

[5 rows x 21 columns]

In [67]:
df_genres_pivot.fillna(0)

value      id  Action  Adventure  Animation  Comedy  Crime  Documentary  \
0           2     0.0        0.0        0.0     0.0    1.0          0.0   
1           3     0.0        0.0        0.0     1.0    0.0          0.0   
2           5     0.0        0.0        0.0     1.0    1.0          0.0   
3           6     1.0        0.0        0.0     0.0    1.0          0.0   
4          11     1.0        1.0        0.0     0.0    0.0          0.0   
...       ...     ...        ...        ...     ...    ...          ...   
42986  465044     0.0        0.0        0.0     0.0    0.0          0.0   
42987  467731     0.0        0.0        0.0     0.0    0.0          0.0   
42988  468343     0.0        0.0        0.0     0.0    0.0          0.0   
42989  468707     0.0        0.0        0.0     1.0    0.0          0.0   
42990  469172     0.0        0.0        0.0     0.0    0.0          0.0   

value  Drama  Family  Fantasy  ...  History  Horror  Music  Mystery  Romance  \
0        1.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
1        1.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
2        0.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
3        0.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
4        0.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
...      ...     ...      ...  ...      ...     ...    ...      ...      ...   
42986    1.0     0.0      1.0  ...      0.0     0.0    0.0      0.0      0.0   
42987    1.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      0.0   
42988    1.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      1.0   
42989    0.0     0.0      0.0  ...      0.0     0.0    0.0      0.0      1.0   
42990    1.0     0.0      1.0  ...      0.0     0.0    0.0      0.0      0.0   

value  ScienceFiction  TVMovie  Thriller  War  Western  
0                 0.0      0.0       0.0  0.0      0.0  
1                 0.0      0.0       0.0  0.0      0.0  
2                 0.0      0.0       0.0  0.0      0.0  
3                 0.0      0.0       1.0  0.0      0.0  
4                 1.0      0.0       0.0  0.0      0.0  
...               ...      ...       ...  ...      ...  
42986             0.0      0.0       0.0  0.0      0.0  
42987             0.0      0.0       0.0  0.0      0.0  
42988             0.0      0.0       0.0  0.0      0.0  
42989             0.0      0.0       0.0  0.0      0.0  
42990             0.0      0.0       0.0  0.0      0.0  

[42991 rows x 21 columns]

In [69]:
# in this section we converted df_genres into a pivot table by metlting the data and adding a new column. We also made sure to fill all null values

### Section 5b: Filter df_movies based on comedies

In [73]:
df_movies.head()

id    imdb_id  adult    budget original_language  \
0    862  tt0114709  False  30000000                en   
1   8844  tt0113497  False  65000000                en   
2  15602  tt0113228  False         0                en   
3  31357  tt0114885  False  16000000                en   
4  11862  tt0113041  False         0                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  popularity  \
0  Led by Woody, Andy's toys live happily in his ...   21.946943   
1  When siblings Judy and Peter discover an encha...   17.015539   
2  A family wedding reignites the ancient feud be...   11.712900   
3  Cheated on, mistreated and stepped on, the wom...    3.859495   
4  Just when George Banks has recovered from his ...    8.387519   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...     10/30/95   
1  [{'iso_3166_1': 'US', 'name': 'United States o...     12/15/95   
2  [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95   
3  [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95   
4  [{'iso_3166_1': 'US', 'name': 'United States o...      2/10/95   

       revenue  runtime                                            tagline  \
0  373554033.0     81.0                                                NaN   
1  262797249.0    104.0          Roll the dice and unleash the excitement!   
2          0.0    101.0  Still Yelling. Still Fighting. Still Ready for...   
3   81452156.0    127.0  Friends are the people who let you be yourself...   
4   76578911.0    106.0  Just When His World Is Back To Normal... He's ...   

                         title  video  vote_average  vote_count  
0                    Toy Story  False           7.7      5415.0  
1                      Jumanji  False           6.9      2413.0  
2             Grumpier Old Men  False           6.5        92.0  
3            Waiting to Exhale  False           6.1        34.0  
4  Father of the Bride Part II  False           5.7       173.0

In [75]:
df_movies_genre = df_movies.merge(df_genres_pivot, how='left', on='id')

In [160]:
df_movies_genre

id    imdb_id  adult    budget original_language  \
0         862  tt0114709  False  30000000                en   
1        8844  tt0113497  False  65000000                en   
2       15602  tt0113228  False         0                en   
3       31357  tt0114885  False  16000000                en   
4       11862  tt0113041  False         0                en   
...       ...        ...    ...       ...               ...   
45458   35731  tt1161951   True         0                en   
45459   55774  tt1153101   True         0                en   
45460   39922  tt0070696   True         0                da   
45461  376004  tt4044464   True    750000                en   
45462  448290  tt2697688   True         0                en   

                    original_title  \
0                        Toy Story   
1                          Jumanji   
2                 Grumpier Old Men   
3                Waiting to Exhale   
4      Father of the Bride Part II   
...                            ...   
45458   Amateur Porn Star Killer 2   
45459                     The Band   
45460                      Dværgen   
45461                   Adulterers   
45462                   Half -Life   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...   21.946943   
1      When siblings Judy and Peter discover an encha...   17.015539   
2      A family wedding reignites the ancient feud be...   11.712900   
3      Cheated on, mistreated and stepped on, the wom...    3.859495   
4      Just when George Banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45458  Shane Ryan's sequel to the disturbing Amateur ...    0.587751   
45459  Australian film about a fictional sub-par Aust...    0.417214   
45460  Olaf and his mother run a boarding house and a...    1.182635   
45461  A man who returns home to find his wife cheati...    1.133770   
45462  An American scientific research facility opens...    0.282680   

                                    production_countries release_date  ...  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...     10/30/95  ...   
1      [{'iso_3166_1': 'US', 'name': 'United States o...     12/15/95  ...   
2      [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95  ...   
3      [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95  ...   
4      [{'iso_3166_1': 'US', 'name': 'United States o...      2/10/95  ...   
...                                                  ...          ...  ...   
45458  [{'iso_3166_1': 'US', 'name': 'United States o...      5/13/08  ...   
45459        [{'iso_3166_1': 'AU', 'name': 'Australia'}]     11/17/09  ...   
45460  [{'iso_3166_1': 'DK', 'name': 'Denmark'}, {'is...     12/20/73  ...   
45461                                                 []       1/5/16  ...   
45462  [{'iso_3166_1': 'US', 'name': 'United States o...          NaN  ...   

       History  Horror Music Mystery Romance  ScienceFiction  TVMovie  \
0          NaN     NaN   NaN     NaN     NaN             NaN      NaN   
1          NaN     NaN   NaN     NaN     NaN             NaN      NaN   
2          NaN     NaN   NaN     NaN     1.0             NaN      NaN   
3          NaN     NaN   NaN     NaN     1.0             NaN      NaN   
4          NaN     NaN   NaN     NaN     NaN             NaN      NaN   
...        ...     ...   ...     ...     ...             ...      ...   
45458      NaN     1.0   NaN     NaN     NaN             NaN      NaN   
45459      NaN     NaN   1.0     NaN     1.0             NaN      NaN   
45460      NaN     1.0   NaN     NaN     NaN             NaN      NaN   
45461      NaN     NaN   NaN     NaN     NaN             NaN      NaN   
45462      NaN     NaN   NaN     NaN     NaN             1.0      NaN   

       Thriller  War  Western  
0           NaN  NaN      NaN  
1           NaN  NaN      NaN  
2           NaN  NaN  

In [172]:
df_movies_comedy = df_movies_genre[df_movies_genre['Comedy'] == 1]
df_movies_comedy

id    imdb_id  adult    budget original_language  \
0         862  tt0114709  False  30000000                en   
2       15602  tt0113228  False         0                en   
3       31357  tt0114885  False  16000000                en   
4       11862  tt0113041  False         0                en   
6       11860  tt0114319  False  58000000                en   
...       ...        ...    ...       ...               ...   
45439  106807  tt0135571  False         0                fr   
45442  420346  tt4130180  False         0                en   
45456   44781  tt0322232   True         0                cn   
45457  324230  tt3094816   True         0                es   
45459   55774  tt1153101   True         0                en   

                       original_title  \
0                           Toy Story   
2                    Grumpier Old Men   
3                   Waiting to Exhale   
4         Father of the Bride Part II   
6                             Sabrina   
...                               ...   
45439  Nouvelles luttes extravagantes   
45442               The Morning After   
45456                           發電悄嬌娃   
45457                     Diet of Sex   
45459                        The Band   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...   21.946943   
2      A family wedding reignites the ancient feud be...   11.712900   
3      Cheated on, mistreated and stepped on, the wom...    3.859495   
4      Just when George Banks has recovered from his ...    8.387519   
6      An ugly duckling having undergone a remarkable...    6.677277   
...                                                  ...         ...   
45439         A series of fantastical wrestling matches.    0.225432   
45442  The Morning After is a feature film that consi...    0.139936   
45456  Electrical Girl centers around a horny young w...    0.001803   
45457  Ágata suffers from a psychological disorder wh...    6.889832   
45459  Australian film about a fictional sub-par Aust...    0.417214   

                                    production_countries release_date  ...  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...     10/30/95  ...   
2      [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95  ...   
3      [{'iso_3166_1': 'US', 'name': 'United States o...     12/22/95  ...   
4      [{'iso_3166_1': 'US', 'name': 'United States o...      2/10/95  ...   
6      [{'iso_3166_1': 'DE', 'name': 'Germany'}, {'is...     12/15/95  ...   
...                                                  ...          ...  ...   
45439           [{'iso_3166_1': 'FR', 'name': 'France'}]       1/1/00  ...   
45442  [{'iso_3166_1': 'US', 'name': 'United States o...      1/11/15  ...   
45456        [{'iso_3166_1': 'HK', 'name': 'Hong Kong'}]      4/26/01  ...   
45457            [{'iso_3166_1': 'ES', 'name': 'Spain'}]      2/14/14  ...   
45459        [{'iso_3166_1': 'AU', 'name': 'Australia'}]     11/17/09  ...   

       History  Horror Music Mystery Romance  ScienceFiction  TVMovie  \
0          NaN     NaN   NaN     NaN     NaN             NaN      NaN   
2          NaN     NaN   NaN     NaN     1.0             NaN      NaN   
3          NaN     NaN   NaN     NaN     1.0             NaN      NaN   
4          NaN     NaN   NaN     NaN     NaN             NaN      NaN   
6          NaN     NaN   NaN     NaN     1.0             NaN      NaN   
...        ...     ...   ...     ...     ...             ...      ...   
45439      NaN     NaN   NaN     NaN     NaN             NaN      NaN   
45442      NaN     NaN   NaN     NaN     NaN             NaN      NaN   
45456      NaN     NaN   NaN     NaN     NaN             NaN      NaN   
45457      NaN     NaN   NaN     NaN     1.0             NaN      NaN   
45459      NaN     NaN   1.0     NaN     1.0             NaN      NaN   

       Thriller  War  Western  
0           NaN  NaN      NaN  
2           NaN  N

In [174]:
df_movies_comedy = df_movies_comedy[['id', 'title']]

In [176]:
df_movies_comedy

id                             title
0         862                         Toy Story
2       15602                  Grumpier Old Men
3       31357                 Waiting to Exhale
4       11862       Father of the Bride Part II
6       11860                           Sabrina
...       ...                               ...
45439  106807  The Fat and Lean Wrestling Match
45442  420346                 The Morning After
45456   44781                   Electrical Girl
45457  324230                       Diet of Sex
45459   55774                          The Band

[13182 rows x 2 columns]

In [178]:
df_ratings

userId  movieId  rating   timestamp
0              1      110     1.0  1425941529
1              1      147     4.5  1425942435
2              1      858     5.0  1425941523
3              1     1221     5.0  1425941546
4              1     1246     5.0  1425941556
...          ...      ...     ...         ...
26024284  270896    58559     5.0  1257031564
26024285  270896    60069     5.0  1257032032
26024286  270896    63082     4.5  1257031764
26024287  270896    64957     4.5  1257033990
26024288  270896    71878     2.0  1257031858

[26024289 rows x 4 columns]

In [180]:
df_comedies_rec = pd.merge(df_movies_comedy,df_ratings,left_on='id',right_on='movieId',how='inner')
df_comedies_rec.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3247720 entries, 0 to 3247719
Data columns (total 6 columns):
 #   Column     Dtype  
---  ------     -----  
 0   id         int64  
 1   title      object 
 2   userId     int64  
 3   movieId    int64  
 4   rating     float64
 5   timestamp  int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 148.7+ MB


In [181]:
# in this section I created multiple dataframes and merged them to filter movies based on them being comedies

## 6. Create a Recommender System using PCA

### Section 6a: Create a pivot table named df_comedies_pivot

#### **OBJECTIVE**: To create the first recommender system, we will first run PCA to identify the number of components to include when creating our correlation matrix. (This is similar to HW03, but first we need to create a pivot table version of the users and movies similar to Figure 8.)

In [193]:
df_comedies_pivot = pd.pivot_table(df_comedies_rec,index='title',columns='userId',values='rating')

In [195]:
df_comedies_pivot =df_comedies_pivot.fillna(0)
df_comedies_pivot.head()

userId                                1       2       3       4       5       \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   

userId                                6       7       8       9       10      \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   

userId                                ...  270887  270888  270889  270890  \
title                                 ...                                   
$5 a Day                              ...     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter  ...     0.0     0.0     0.0     0.0   
10 Items or Less                      ...     0.0     0.0     0.0     0.0   
10 Things I Hate About You            ...     0.0     0.0     0.0     0.0   
12 Dates of Christmas                 ...     0.0     0.0     0.0     0.0   

userId                                270891  270892  270893  270894  270895  \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   

userId                                270896  
title                                         
$5 a Day                                 0.0  
00 Schneider - Jagd auf Nihil Baxter     0.0  
10 Items or Less                         0.0  
10 Things I Hate About You               0.0  
12 Dates of Christmas                    0.0  

[5 rows x 243237 columns]

In [197]:
comedy_names = df_comedies_pivot.index
comedy_names

Index(['$5 a Day', '00 Schneider - Jagd auf Nihil Baxter', '10 Items or Less',
       '10 Things I Hate About You', '12 Dates of Christmas',
       '12 Wishes of Christmas', '2 Days in Paris', '2001: A Space Travesty',
       '24 Hour Party People', '27 Dresses',
       ...
       'Zapped Again!', 'Zazie dans le métro', 'Zero Charisma',
       'Zero for Conduct', 'Zigzag of Success', '¡A volar joven!',
       '¡Three Amigos!', 'Ödipussi', 'Şaban Oğlu Şaban',
       'Šíleně smutná princezna'],
      dtype='object', name='title', length=2098)

In [ ]:
# In this section I created a picot table named df_comedies_pivot

### Section 6b: Create a PCA for df_comedy_pivot

#### **OBJECTIVE**: Time to reduce the number of dimensions from 243,237 (number of columns in df_comedy_pivot) to a more reasonable number that still captures the explained variance.

In [205]:
from sklearn.decomposition import PCA

n_pcs = 85
model = PCA(n_components=n_pcs)

# Fit the model - return projected data
df_comedy_pca = model.fit_transform(df_comedies_pivot)

#Create a DataFrame of the projected data
df_comedy_pca = pd.DataFrame(df_comedy_pca)
df_comedy_pca.head()

0          1          2          3          4         5          6   \
0 -17.316626   0.499356   2.333956   1.962810   2.024633  0.201153  -0.809383   
1 -18.377745   0.164179   3.325374   3.355762   1.882705 -0.308388  -0.951306   
2  17.967140 -11.241489 -31.784518  16.958452 -15.103681 -6.659356 -19.224768   
3   2.658676  -2.767690  -3.362124 -10.739989   0.516089  6.207526   1.301159   
4 -19.480163   0.492870   4.316852   3.361627   2.496690 -0.513061  -1.032770   

         7         8         9   ...        75        76        77        78  \
0  0.403978 -0.390762  1.178846  ...  0.138228  0.208554 -0.330598  0.324573   
1  0.073027  0.596650  1.576595  ...  0.497895  0.208299  0.226417  0.104311   
2 -4.902429 -3.736171 -4.888509  ...  5.466262 -0.559186 -1.677183 -2.110319   
3  0.738756 -3.232905 -2.295874  ...  1.248045 -1.608962 -0.922707  3.342986   
4  0.573200  0.654861  1.769396  ...  0.345011  0.363111  0.232618 -0.088784   

         79        80         81        82         83        84  
0  0.206814  0.268084  -0.047831  0.079492   0.393863  0.150498  
1  0.322652  0.390428  -0.034523  0.141751  -0.358892  0.132177  
2 -2.802096  4.689892 -11.198520 -7.784505 -16.653218 -3.336050  
3 -0.200344 -0.390274   3.404877  0.953849   0.348109 -3.832525  
4  0.165555 -0.030790  -0.126661 -0.084140   0.204708  0.307816  

[5 rows x 85 columns]

In [206]:
names = []
for x in range(n_pcs):
    names.append('PC%d'%(x+1))

names

['PC1',
 'PC2',
 'PC3',
 'PC4',
 'PC5',
 'PC6',
 'PC7',
 'PC8',
 'PC9',
 'PC10',
 'PC11',
 'PC12',
 'PC13',
 'PC14',
 'PC15',
 'PC16',
 'PC17',
 'PC18',
 'PC19',
 'PC20',
 'PC21',
 'PC22',
 'PC23',
 'PC24',
 'PC25',
 'PC26',
 'PC27',
 'PC28',
 'PC29',
 'PC30',
 'PC31',
 'PC32',
 'PC33',
 'PC34',
 'PC35',
 'PC36',
 'PC37',
 'PC38',
 'PC39',
 'PC40',
 'PC41',
 'PC42',
 'PC43',
 'PC44',
 'PC45',
 'PC46',
 'PC47',
 'PC48',
 'PC49',
 'PC50',
 'PC51',
 'PC52',
 'PC53',
 'PC54',
 'PC55',
 'PC56',
 'PC57',
 'PC58',
 'PC59',
 'PC60',
 'PC61',
 'PC62',
 'PC63',
 'PC64',
 'PC65',
 'PC66',
 'PC67',
 'PC68',
 'PC69',
 'PC70',
 'PC71',
 'PC72',
 'PC73',
 'PC74',
 'PC75',
 'PC76',
 'PC77',
 'PC78',
 'PC79',
 'PC80',
 'PC81',
 'PC82',
 'PC83',
 'PC84',
 'PC85']

In [209]:
df_comedy_pca.columns = names
df_comedy_pca

PC1        PC2        PC3        PC4        PC5       PC6  \
0    -17.316626   0.499356   2.333956   1.962810   2.024633  0.201153   
1    -18.377745   0.164179   3.325374   3.355762   1.882705 -0.308388   
2     17.967140 -11.241489 -31.784518  16.958452 -15.103681 -6.659356   
3      2.658676  -2.767690  -3.362124 -10.739989   0.516089  6.207526   
4    -19.480163   0.492870   4.316852   3.361627   2.496690 -0.513061   
...         ...        ...        ...        ...        ...       ...   
2093 -12.342673  -3.330853   3.922789   1.391081  -3.685394 -2.984989   
2094 -18.882186   0.437203   3.812984   3.067699   2.203784 -0.820199   
2095 -15.283326  -1.661003   2.654492   3.091673  -0.650131  0.218650   
2096 -17.166576  -0.939541   3.657008   3.720439   0.447326 -0.600584   
2097 -13.165311   0.608230  -0.450099  -1.508189   0.976331  1.659172   

            PC7       PC8       PC9      PC10  ...      PC76      PC77  \
0     -0.809383  0.403978 -0.390762  1.178846  ...  0.138228  0.208554   
1     -0.951306  0.073027  0.596650  1.576595  ...  0.497895  0.208299   
2    -19.224768 -4.902429 -3.736171 -4.888509  ...  5.466262 -0.559186   
3      1.301159  0.738756 -3.232905 -2.295874  ...  1.248045 -1.608962   
4     -1.032770  0.573200  0.654861  1.769396  ...  0.345011  0.363111   
...         ...       ...       ...       ...  ...       ...       ...   
2093  -2.550258  0.152643  1.483567 -3.607085  ... -1.523807 -0.438140   
2094  -0.998774  0.367577  0.526603  1.507941  ...  0.314823  0.417458   
2095   0.168010 -2.634645  1.204786  0.877996  ... -0.345862 -1.133823   
2096  -0.579884 -1.434289  1.475163  0.867067  ... -0.081847 -0.171739   
2097  -1.128863  1.327382 -2.385064 -0.079937  ... -0.477951 -0.469359   

          PC78      PC79      PC80      PC81       PC82      PC83       PC84  \
0    -0.330598  0.324573  0.206814  0.268084  -0.047831  0.079492   0.393863   
1     0.226417  0.104311  0.322652  0.390428  -0.034523  0.141751  -0.358892   
2    -1.677183 -2.110319 -2.802096  4.689892 -11.198520 -7.784505 -16.653218   
3    -0.922707  3.342986 -0.200344 -0.390274   3.404877  0.953849   0.348109   
4     0.232618 -0.088784  0.165555 -0.030790  -0.126661 -0.084140   0.204708   
...        ...       ...       ...       ...        ...       ...        ...   
2093  0.127665 -1.657129 -1.048897 -0.458853  -0.092962 -1.116630  -2.029188   
2094  0.206948  0.161236  0.412671 -0.055970  -0.178663 -0.048276   0.170014   
2095  1.216430  0.056451 -0.303272 -0.306546   0.982349  0.013022   0.453319   
2096  0.884079  0.267353  0.003752  0.120089   0.531274  0.078322   0.093615   
2097 -1.494422  0.165382  0.114438  0.523378  -0.350411  0.585610   0.164864   

          PC85  
0     0.150498  
1     0.132177  
2    -3.336050  
3    -3.832525  
4     0.307816  
...        ...  
2093  0.038486  
2094  0.383029  
2095 -0.081056  
2096  0.520879  
2097 -0.369382  

[2098 rows x 85 columns]

In [211]:
explained_ratio = model.explained_variance_ratio_
explained_ratio = pd.DataFrame(explained_ratio).reset_index()
explained_ratio = explained_ratio.rename(columns = {0: 'explained_ratio', 'index': 'PC'})
explained_ratio['explained_ratio_sum']= np.cumsum(explained_ratio.explained_ratio)
explained_ratio

PC  explained_ratio  explained_ratio_sum
0    0         0.194278             0.194278
1    1         0.053022             0.247300
2    2         0.032253             0.279553
3    3         0.023536             0.303089
4    4         0.017345             0.320434
..  ..              ...                  ...
80  80         0.002335             0.692454
81  81         0.002327             0.694781
82  82         0.002310             0.697090
83  83         0.002285             0.699376
84  84         0.002241             0.701617

[85 rows x 3 columns]

In [213]:
# In this section I created a PCA for df_comedy_pivot using n_pcs of 5 and then creates a datafram eto show the explained_ratio & explained_ratio_sum
# this explains 70% of variance and a gerneal rule form DR.A is to use that much so you don not lose too much of the variance explained

### Section 6c: Create a Correlation Matrix for a Recommender System

In [233]:


corr_mat_comedy = np.corrcoef(df_comedy_pca)
corr_mat_comedy  = pd.DataFrame(corr_mat_comedy, columns=(comedy_names))
corr_mat_comedy.index = comedy_names
corr_mat_comedy

title                                 $5 a Day  \
title                                            
$5 a Day                              1.000000   
00 Schneider - Jagd auf Nihil Baxter  0.971634   
10 Items or Less                     -0.373993   
10 Things I Hate About You           -0.165291   
12 Dates of Christmas                 0.973867   
...                                        ...   
¡A volar joven!                       0.573452   
¡Three Amigos!                        0.978543   
Ödipussi                              0.853584   
Şaban Oğlu Şaban                      0.922739   
Šíleně smutná princezna               0.847386   

title                                 00 Schneider - Jagd auf Nihil Baxter  \
title                                                                        
$5 a Day                                                          0.971634   
00 Schneider - Jagd auf Nihil Baxter                              1.000000   
10 Items or Less                                                 -0.304425   
10 Things I Hate About You                                       -0.279442   
12 Dates of Christmas                                             0.994643   
...                                                                    ...   
¡A volar joven!                                                   0.614316   
¡Three Amigos!                                                    0.994265   
Ödipussi                                                          0.890434   
Şaban Oğlu Şaban                                                  0.965733   
Šíleně smutná princezna                                           0.738915   

title                                 10 Items or Less  \
title                                                    
$5 a Day                                     -0.373993   
00 Schneider - Jagd auf Nihil Baxter         -0.304425   
10 Items or Less                              1.000000   
10 Things I Hate About You                   -0.066793   
12 Dates of Christmas                        -0.361840   
...                                                ...   
¡A volar joven!                              -0.023021   
¡Three Amigos!                               -0.348774   
Ödipussi                                     -0.209252   
Şaban Oğlu Şaban                             -0.268616   
Šíleně smutná princezna                      -0.284148   

title                                 10 Things I Hate About You  \
title                                                              
$5 a Day                                               -0.165291   
00 Schneider - Jagd auf Nihil Baxter                   -0.279442   
10 Items or Less                                       -0.066793   
10 Things I Hate About You                              1.000000   
12 Dates of Christmas                                  -0.291513   
...                                                          ...   
¡A volar joven!                                        -0.185530   
¡Three Amigos!                                         -0.279948   
Ödipussi                                               -0.127305   
Şaban Oğlu Şaban                                       -0.267189   
Šíleně smutná princezna                                 0.196052   

title                                 12 Dates of Christmas  \
title                                                         
$5 a Day                                           0.973867   
00 Schneider - Jagd auf Nihil Baxter               0.994643   
10 Items or Less                                  -0.361840   
10 Things I Hate About You                        -0.291513   
12 Dates of Christmas                              1.000000   
...                                                     ...   
¡A volar joven!                                    0.593748   
¡Three Amigos!                                     0.998276   
Ödipussi                                           0.862770   
Ş

In [272]:
corr_ToyStory= pd.DataFrame(corr_mat_comedy['Toy Story'])
corr_ToyStory.shape

(2098, 1)

In [270]:
corr_ToyStory.sort_values(ascending=False, by='Toy Story').head(10)

Toy Story
title                                                        
Toy Story                                            1.000000
Pretty Woman                                         0.934451
Borat: Cultural Learnings of America for Make B...   0.930589
Party Monster                                        0.927196
The 51st State                                       0.918242
The Front Page                                       0.913665
Guess Who's Coming to Dinner                         0.905004
Miss Congeniality                                    0.902775
Armed and Dangerous                                  0.893233
Life of Brian                                        0.890926

In [245]:
# In this section I created a correlation Matrix for a recommendar system then created a new dataframe for toy story and and sorted the values with the largest correlation to toy story

### Section 6d: Explain the results for the Recommender System for Toy Story

#### The results I'm seeing is showing the top 10 comedy movies with the highest correlation to Toy Story after creating our PCA dataframe to limit the amount of varibles needed to explain the data. These movies listed are great movies, now when it comes to them making sense to be with toy story is not reasonible. Most of these movies are rated R or have huge controversy. When I think of Toy Story Comedy I'm thinking PG movies, not PG-13 to R to very controversial.These are some great movies, but not good choices when it comes to reccomendations after Toy Story.

In [ ]:
#In this section I explained the results to the Reccomender System based on the correlation matrix and PCA.

## 7. Create a Recommender System using SVD

### Section 7a: Create a correlation matrix from SVD

#### **OBJECTIVE**: For some reason, I just do not believe the results from the PCA. Why would this be the case? If you go back to the Scenarios/Presentations, you will see that one method for Recommender Systems does a better job than the other when the data is sparse. (Sparse refers to data where most of the values are NaNs. Since most users only rate a few movies, most of the original data in corr_comedy_pivot (prior to filling in NaNs with 0) were NaN. Therefore, we should look at SVD as a better option for a recommender system.)

In [295]:
from sklearn.decomposition import TruncatedSVD
SVD = TruncatedSVD(n_components=85)
resultant_matrix_SVD = SVD.fit_transform(df_comedies_pivot)
resultant_matrix_SVD.shape

(2098, 85)

In [296]:
df_resultant_matrix_SVD = pd.DataFrame(resultant_matrix_SVD)
df_resultant_matrix_SVD.head()

0          1          2          3          4         5          6   \
0   2.396412  -0.035689  -2.275408  -1.429823  -0.586303  0.686190   0.313951   
1   1.256946  -0.361763  -1.153367  -0.083664  -0.775622  0.174191   0.163621   
2  39.009695 -11.883545 -35.677016  14.850666 -18.009552 -6.504325 -17.181710   
3  23.020943  -3.334630  -8.907074 -13.871254  -1.567059  6.625123   2.362064   
4   0.056788  -0.023487  -0.069409  -0.032385  -0.085092 -0.035982   0.016413   

         7         8         9   ...        75        76        77        78  \
0 -0.287220 -1.250557 -0.455165  ...  0.062082  0.171764  0.108670  0.401240   
1 -0.598599 -0.191860 -0.328065  ...  0.434836 -0.021123  0.320325  0.318570   
2 -7.090466 -7.623928 -7.970909  ...  8.670033  4.361481  4.237440  4.712288   
3  0.093609 -4.782399 -2.946071  ...  0.164293 -0.775422  2.344925  2.291163   
4 -0.070980 -0.029911 -0.061906  ...  0.023457 -0.007511  0.031649  0.016874   

          79        80        81        82        83        84  
0   0.455806  0.057260 -0.111759 -0.074853 -0.633217  0.024190  
1   0.211500  0.256018  0.103394  0.147585 -0.201341 -0.037846  
2 -13.743174  8.252982 -0.207122  8.888949 -7.736211 -2.626811  
3   1.635762 -1.781341 -0.309417 -2.148647 -2.018895 -1.216566  
4   0.006400  0.020340 -0.006049  0.060943 -0.019388  0.003965  

[5 rows x 85 columns]

In [297]:
corr_mat_SVD_Comedy = np.corrcoef(resultant_matrix_SVD)
corr_mat_SVD_Comedy = pd.DataFrame(corr_mat_SVD_Comedy, columns=(comedy_names))
corr_mat_SVD_Comedy.index = comedy_names
corr_mat_SVD_Comedy

title                                 $5 a Day  \
title                                            
$5 a Day                              1.000000   
00 Schneider - Jagd auf Nihil Baxter  0.592499   
10 Items or Less                      0.370557   
10 Things I Hate About You            0.779952   
12 Dates of Christmas                 0.531311   
...                                        ...   
¡A volar joven!                       0.462118   
¡Three Amigos!                        0.748964   
Ödipussi                              0.517836   
Şaban Oğlu Şaban                      0.385410   
Šíleně smutná princezna               0.917952   

title                                 00 Schneider - Jagd auf Nihil Baxter  \
title                                                                        
$5 a Day                                                          0.592499   
00 Schneider - Jagd auf Nihil Baxter                              1.000000   
10 Items or Less                                                  0.715147   
10 Things I Hate About You                                        0.565884   
12 Dates of Christmas                                             0.779999   
...                                                                    ...   
¡A volar joven!                                                   0.646922   
¡Three Amigos!                                                    0.690063   
Ödipussi                                                          0.810017   
Şaban Oğlu Şaban                                                  0.772763   
Šíleně smutná princezna                                           0.575811   

title                                 10 Items or Less  \
title                                                    
$5 a Day                                      0.370557   
00 Schneider - Jagd auf Nihil Baxter          0.715147   
10 Items or Less                              1.000000   
10 Things I Hate About You                    0.388474   
12 Dates of Christmas                         0.473830   
...                                                ...   
¡A volar joven!                               0.516594   
¡Three Amigos!                                0.558165   
Ödipussi                                      0.497266   
Şaban Oğlu Şaban                              0.494851   
Šíleně smutná princezna                       0.441682   

title                                 10 Things I Hate About You  \
title                                                              
$5 a Day                                                0.779952   
00 Schneider - Jagd auf Nihil Baxter                    0.565884   
10 Items or Less                                        0.388474   
10 Things I Hate About You                              1.000000   
12 Dates of Christmas                                   0.412744   
...                                                          ...   
¡A volar joven!                                         0.492518   
¡Three Amigos!                                          0.625369   
Ödipussi                                                0.577339   
Şaban Oğlu Şaban                                        0.442536   
Šíleně smutná princezna                                 0.828079   

title                                 12 Dates of Christmas  \
title                                                         
$5 a Day                                           0.531311   
00 Schneider - Jagd auf Nihil Baxter               0.779999   
10 Items or Less                                   0.473830   
10 Things I Hate About You                         0.412744   
12 Dates of Christmas                              1.000000   
...                                                     ...   
¡A volar joven!                                    0.705858   
¡Three Amigos!                                     0.742442   
Ödipussi                                           0.762472   
Ş

In [298]:
corr_SVD_ToyStory= pd.DataFrame(corr_mat_SVD_Comedy['Click'])
corr_SVD_ToyStory.shape

(2098, 1)

In [299]:
corr_SVD_ToyStory.sort_values(ascending=False, by='Toy Story').head(10)

Toy Story
title                                                        
Toy Story                                            1.000000
Pretty Woman                                         0.927157
Wild at Heart                                        0.907211
Miss Congeniality                                    0.905439
Army of Darkness                                     0.897714
Party Monster                                        0.893718
Borat: Cultural Learnings of America for Make B...   0.890019
Shower                                               0.886602
Junebug                                              0.885568
The Front Page                                       0.883632

In [ ]:
# In this section I created a Reccomedner system from the correlation Matrix from SVD

### Section 7b: Explain the results for the Recommender System using SVD for Toy Story

#### Again similar to my last explaination over the recccomender system I expected there to be more childrens movies reccomended and less rated R comedies. These movies for my age look like a great reccomendation, but for anyone who was younger it would seem these movies shouldn't be reccomended, but these are great movies. I guess if we wanted to go further we could make it to where it was for children only if we wanted to spin it that way. But, all of these movies are very highly rated and I can see why they would be reccomended next to toy story. I think compared to the first results they are very similar with only a handful of movies changed. I would say it is hard to say which of these results are better, but I think they both do a great job when it comes to great movies being reccomended, for the sake of the homework I will say the SVD system would be better due to the sparse data we had.

In [ ]:
# explainatiojn of the results from the SVD reccomender system 

## 8. Create a Recommender System for your favorite

#### **OBJECTIVE**: Now that you have created a PCA and a SVD Recommender System for Toy Story. Select any comedy that you consider one of your favorites to conduct a recommender system. Pick either SVD or PCA (not both) based on your analysis in Section 7b.

### Section 8a: Create your Recommender System for your favorite movie.

In [309]:
df_comedies_pivot

userId                                1       2       3       4       5       \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   
...                                      ...     ...     ...     ...     ...   
¡A volar joven!                          0.0     0.0     0.0     0.0     0.0   
¡Three Amigos!                           0.0     0.0     0.0     0.0     0.0   
Ödipussi                                 0.0     0.0     0.0     0.0     0.0   
Şaban Oğlu Şaban                         0.0     0.0     0.0     0.0     0.0   
Šíleně smutná princezna                  0.0     0.0     0.0     0.0     0.0   

userId                                6       7       8       9       10      \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   
...                                      ...     ...     ...     ...     ...   
¡A volar joven!                          0.0     0.0     0.0     0.0     0.0   
¡Three Amigos!                           0.0     0.0     0.0     0.0     0.0   
Ödipussi                                 0.0     0.0     0.0     0.0     0.0   
Şaban Oğlu Şaban                         0.0     0.0     0.0     0.0     0.0   
Šíleně smutná princezna                  0.0     0.0     0.0     0.0     0.0   

userId                                ...  270887  270888  270889  270890  \
title                                 ...                                   
$5 a Day                              ...     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter  ...     0.0     0.0     0.0     0.0   
10 Items or Less                      ...     0.0     0.0     0.0     0.0   
10 Things I Hate About You            ...     0.0     0.0     0.0     0.0   
12 Dates of Christmas                 ...     0.0     0.0     0.0     0.0   
...                                   ...     ...     ...     ...     ...   
¡A volar joven!                       ...     0.0     0.0     0.0     0.0   
¡Three Amigos!                        ...     0.0     0.0     0.0     0.0   
Ödipussi                              ...     0.0     0.0     0.0     0.0   
Şaban Oğlu Şaban                      ...     0.0     0.0     0.0     0.0   
Šíleně smutná princezna               ...     0.0     0.0     0.0     0.0   

userId                                270891  270892  270893  270894  270895  \
title                                                                          
$5 a Day                                 0.0     0.0     0.0     0.0     0.0   
00 Schneider - Jagd auf Nihil Baxter     0.0     0.0     0.0     0.0     0.0   
10 Items or Less                         0.0     0.0     0.0     0.0     0.0   
10 Things I Hate About You               0.0     0.0     0.0     0.0     0.0   
12 Dates of Christmas                    0.0     0.0     0.0     0.0     0.0   
...                                      ...     ...     ...     ...     ...   
¡A volar joven!                          0.0     0.0     0.0     0.0     0.0   
¡Three Amigos!                           0.0     0.0     0.0     0.0     0.0   
Ödipussi                                 0.0     0.0     0.0     0.0     0.0   
Şaban Oğlu Şaban                    

In [313]:
comedy_movie_names = df_comedies_pivot.index
comedy_movie_names

Index(['$5 a Day', '00 Schneider - Jagd auf Nihil Baxter', '10 Items or Less',
       '10 Things I Hate About You', '12 Dates of Christmas',
       '12 Wishes of Christmas', '2 Days in Paris', '2001: A Space Travesty',
       '24 Hour Party People', '27 Dresses',
       ...
       'Zapped Again!', 'Zazie dans le métro', 'Zero Charisma',
       'Zero for Conduct', 'Zigzag of Success', '¡A volar joven!',
       '¡Three Amigos!', 'Ödipussi', 'Şaban Oğlu Şaban',
       'Šíleně smutná princezna'],
      dtype='object', name='title', length=2098)

In [315]:
SVD = TruncatedSVD(n_components=85)
resultant_matrix_SVD = SVD.fit_transform(df_comedies_pivot)
resultant_matrix_SVD.shape

(2098, 85)

In [316]:
df_resultant_matrix_SVD = pd.DataFrame(resultant_matrix_SVD)
df_resultant_matrix_SVD.head()

0          1          2          3          4         5          6   \
0   2.396412  -0.035689  -2.275408  -1.429823  -0.586308  0.686210   0.313965   
1   1.256946  -0.361763  -1.153367  -0.083663  -0.775622  0.174200   0.163639   
2  39.009695 -11.883545 -35.677013  14.850666 -18.009578 -6.504352 -17.180985   
3  23.020943  -3.334630  -8.907074 -13.871253  -1.567063  6.625214   2.361993   
4   0.056788  -0.023487  -0.069409  -0.032385  -0.085092 -0.035981   0.016415   

         7         8         9   ...        75        76        77        78  \
0 -0.287247 -1.250485 -0.455221  ... -0.182164 -0.486495  0.429124  0.254124   
1 -0.598615 -0.191859 -0.328105  ... -0.102038 -0.100692  0.152054 -0.004353   
2 -7.090245 -7.624454 -7.971843  ...  5.358510 -3.855341 -3.907987 -0.670025   
3  0.093343 -4.782178 -2.946098  ... -2.264510 -1.130004  5.733891  0.099514   
4 -0.070981 -0.029909 -0.061908  ... -0.031469  0.007975  0.015159 -0.021607   

         79        80        81         82        83        84  
0  0.124285 -0.647638  0.213969  -0.315946 -0.029482  0.781500  
1 -0.036035 -0.292478  0.145309  -0.955859 -0.000323  0.125714  
2  2.545839 -0.115801 -8.959502 -14.382403  5.185863 -2.379428  
3  1.883604 -3.399175 -2.089757   0.022088 -0.039633  3.979439  
4  0.018652 -0.037076 -0.038050  -0.052417  0.006275 -0.012054  

[5 rows x 85 columns]

In [319]:
corr_mat_SVD_Favorite = np.corrcoef(resultant_matrix_SVD)
corr_mat_SVD_Favorite = pd.DataFrame(corr_mat_SVD_Comedy, columns=(comedy_movie_names))
corr_mat_SVD_Favorite.index = comedy_movie_names
corr_mat_SVD_Favorite

title                                 $5 a Day  \
title                                            
$5 a Day                              1.000000   
00 Schneider - Jagd auf Nihil Baxter  0.592499   
10 Items or Less                      0.370557   
10 Things I Hate About You            0.779952   
12 Dates of Christmas                 0.531311   
...                                        ...   
¡A volar joven!                       0.462118   
¡Three Amigos!                        0.748964   
Ödipussi                              0.517836   
Şaban Oğlu Şaban                      0.385410   
Šíleně smutná princezna               0.917952   

title                                 00 Schneider - Jagd auf Nihil Baxter  \
title                                                                        
$5 a Day                                                          0.592499   
00 Schneider - Jagd auf Nihil Baxter                              1.000000   
10 Items or Less                                                  0.715147   
10 Things I Hate About You                                        0.565884   
12 Dates of Christmas                                             0.779999   
...                                                                    ...   
¡A volar joven!                                                   0.646922   
¡Three Amigos!                                                    0.690063   
Ödipussi                                                          0.810017   
Şaban Oğlu Şaban                                                  0.772763   
Šíleně smutná princezna                                           0.575811   

title                                 10 Items or Less  \
title                                                    
$5 a Day                                      0.370557   
00 Schneider - Jagd auf Nihil Baxter          0.715147   
10 Items or Less                              1.000000   
10 Things I Hate About You                    0.388474   
12 Dates of Christmas                         0.473830   
...                                                ...   
¡A volar joven!                               0.516594   
¡Three Amigos!                                0.558165   
Ödipussi                                      0.497266   
Şaban Oğlu Şaban                              0.494851   
Šíleně smutná princezna                       0.441682   

title                                 10 Things I Hate About You  \
title                                                              
$5 a Day                                                0.779952   
00 Schneider - Jagd auf Nihil Baxter                    0.565884   
10 Items or Less                                        0.388474   
10 Things I Hate About You                              1.000000   
12 Dates of Christmas                                   0.412744   
...                                                          ...   
¡A volar joven!                                         0.492518   
¡Three Amigos!                                          0.625369   
Ödipussi                                                0.577339   
Şaban Oğlu Şaban                                        0.442536   
Šíleně smutná princezna                                 0.828079   

title                                 12 Dates of Christmas  \
title                                                         
$5 a Day                                           0.531311   
00 Schneider - Jagd auf Nihil Baxter               0.779999   
10 Items or Less                                   0.473830   
10 Things I Hate About You                         0.412744   
12 Dates of Christmas                              1.000000   
...                                                     ...   
¡A volar joven!                                    0.705858   
¡Three Amigos!                                     0.742442   
Ödipussi                                           0.762472   
Ş

In [340]:
corr_SVD_BoratCulturalLearningsofAmericaforMakeBenefitGloriousNationofKazakhstan= pd.DataFrame(corr_mat_SVD_Favorite['Borat: Cultural Learnings of America for Make Benefit Glorious Nation of Kazakhstan'])
corr_SVD_BoratCulturalLearningsofAmericaforMakeBenefitGloriousNationofKazakhstan.shape

(2098, 1)

In [342]:
corr_SVD_BoratCulturalLearningsofAmericaforMakeBenefitGloriousNationofKazakhstan.sort_values(ascending=False, by='Borat: Cultural Learnings of America for Make Benefit Glorious Nation of Kazakhstan').head(10)

Borat: Cultural Learnings of America for Make Benefit Glorious Nation of Kazakhstan
title                                                                                                                                  
Borat: Cultural Learnings of America for Make B...                                           1.000000                                  
Cool Runnings                                                                                0.920666                                  
Party Monster                                                                                0.912207                                  
Wild at Heart                                                                                0.905052                                  
Toy Story                                                                                    0.890019                                  
Berlin is in Germany                                                                         0.889931                                  
Life of Brian                                                                                0.885528                                  
Manhattan                                                                                    0.883483                                  
Army of Darkness                                                                             0.862821                                  
All About My Mother                                                                          0.857777

In [ ]:
# in this sectuon I chose my favorite movie and did the SVD correlation reccomendation matrix

### Section 8b: Explain the results for the Recommender System for your favorite comedy

#### These choices are okay, I will say my favorite comedy was not in the list sadly, so I used one I do love Borat. These do make sense as they were correlated with Toy Story earlier so I can see that all of these movies are very similar and reccomended together. I am probably not going to watch any of these sadly. I googled them and not a big fan of any of these, other than Toy Story and Army of Darkness. I could maybe chose another movie and look, but I'll take my chances on the usual Netflix browse.